src/redis-helper.ts

```diff
@@ -119,11 +176,16 @@ export async function getTopTracksForArtist(redisClient: redis.RedisClient,
     const topTracksFromRedis: RedisTrack[]     = [];
     let   topTracksFromSpotify: SpotifyTrack[] = [];
 
-    if (!artist.top_track_ids || artist.top_track_ids.length === 0) {
-        // We've never gotten tracks and saved their ids for this artist, need to call spotify for tracks,
-        // save their ids for this artist, and save the tracks themselves. We cache all track IDs per artist
-        // and tracks themselves, but only return the number requested
-        console.log(`No top track ids for spotify artist ${
+    const lastUpdated = await getArtistTopTracksLastUpdated(redisClient, artist.id);
+
+    console.log(`Last updated top tracks: ${lastUpdated}`);
+
+    if (!artist.top_track_ids || artist.top_track_ids.length === 0 || !lastUpdated ||
```

Comment:
Can you pre-calc the subtraction you do in a variable before the if-statement please? So the last clause would just be `lastUpdated < refreshThresholdMs` or something, variable name is up to you

Reply:
Done